### PR TITLE
Add code coverage to CI

### PR DESCRIPTION
**Pull request by TomNicholas**
@keewis [pointed out](https://github.com/xarray-contrib/pint-xarray/pull/11#issuecomment-655632247) that after moving to xarray-contrib we seem to have lost the CI. I've re-created the Azure pipeline that's supposed to run the CI tests (including pytest and black), which I have been able to re-run on master. I want to check it's working on new PRs.

### CODE REVIEW

**Comment by TomNicholas** [comment]
Also was there any other integrations we were using? PEP8speaks bot?

**Comment by TomNicholas** [comment]
@keewis I think renaming the package in #15 broke readthedocs too #13

**Comment by keewis** [comment]
yes, I forgot to update the import in `conf.py`. This is fixed in #11, so if / when we merge that, RTD builds will be fixed, too (I could create a new PR for that if necessary, though).

**Comment by keewis** [comment]
I think the only extra integration we had was pep8speaks, although I'm not quite sure how much value this has: `flake8` reports more than just PEP8, and usually the reported errors are fixed by a single run of `black`.

Also, while we're at it: do you think we can investigate the actions integration of pipelines? Right now, only a single CI action is visible (see also https://github.com/xarray-contrib/pint-xarray/pull/11#issuecomment-650755309)

**Comment by TomNicholas** [comment]
> This is fixed in #11

Okay great, thanks.

> I think the only extra integration we had was pep8speaks, although I'm not quite sure how much value this has

It's kind of nice that it comments on the PR says what was wrong with the code, but I agree it's not really adding that much.

> do you think we can investigate the actions integration of pipelines? Right now, only a single CI action is visible

You mean how it just says "xarray-contrib.pint-xarray" instead of listing each test individually? I'm not sure how to change that but I'll have a look.

**Comment by keewis** [comment]
me neither. Let's ping @shoyer since he seems to have been the one to set this up for `xarray`

**Comment by TomNicholas** [comment]
I found [this SO question](https://stackoverflow.com/questions/52374098/how-to-report-individual-jobs-of-pipeline-as-github-checks-for-azure-pipelines) but it doesn't actually tell you how to split them up.

Code Coverage! We should add that too - I'll try and do it now.

**Comment by shoyer** [comment]
I'm not sure why Azure looks different here than for xarray. I don't think that's anything I explicitly set.

There might be an option hidden somewhere in the Azure Pipelines settings?

**Comment by TomNicholas** [comment]
I had a look in the settings but couldn't see anything. I've emailed to commenter on that stack overflow comment who said he works at Azure because why not.

Also @shoyer these Azure pipelines are currently just set up on an account in my name, given access to this repo. Is that the way I'm supposed to be doing it? Can I give other people access too?

@keewis I think I've got codecov working now so if you want to merge this into master we should then have coverage stats.

**Comment by keewis** [comment]
Looks good to me, but before I merge: what's the difference between both coverage checks?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -31,10 +31,15 @@ jobs:
     displayName: 'Install dependencies'
 
   - script: |
-      pip install pytest pytest-azurepipelines
-      pytest --verbose
+      pip install pytest pytest-azurepipelines pytest-cov
```
Comment (keewis) [inline azure-pipelines.yml:34]: would it make sense to add these and all of the linters to `requirements.txt`? Or should we rather maintain a separate CI requirements file?

**Comment by keewis** [comment]
since this is about CI in general: let me know if anyone wants to become a maintainer on RTD

File: azure-pipelines.yml (hunk 1 above)
Comment (keewis) [reply azure-pipelines.yml:34]: Let's do that in a separate PR.

**Comment by keewis** [comment]
I still don't understand the difference between two coverage checks, but let's merge for now.

**Pull request merged**